Modules/xSharePoint/DSCResources/MSFT_xSPFarmSolution/MSFT_xSPFarmSolution.psm1

```diff
@@ -0,0 +1,341 @@
+                    Identity = $solution
+                    GACDeployment = $solution.ContainsGlobalAssembly
+                    Local = $false
+                    Verbose = $false
+                }
+
+                if (!$solution.ContainsWebApplicationResource) 
+                {
+                    Install-SPSolution @runParams
+                }
+                else
+                {
+                    if ($webApps -eq $null -or $webApps.Length -eq 0) 
+                    {
+                        $runParams.Add("AllWebApplications", $true)
+
+                        Install-SPSolution @runParams
+                    }
+                    else
+                    {
+                        foreach ($webApp in $webApps)
+                        {
+                            $runParams["WebApplication"] = $webApp 
+
+                            Install-SPSolution @runParams
+                        }
+                    }
+                }
+
+            }
+        }
+       
+     } 
+
+     WaitFor-SolutionJob -SolutionName $Name
+
+    if ($Ensure -eq "Absent")
+    {
+        $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+            $params = $args[0]
+        
+            $runParams = @{ 
+                Identity = $params.Name
+                Confirm = $false
+                Verbose = $false
+            }
+
+            Remove-SPSolution @runParams
+
+        }
+    }
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [String]   $Name,
+        [parameter(Mandatory = $true)]  [String]   $LiteralPath,
+        [parameter(Mandatory = $false)] [String[]] $WebApplications = @(),
+        [parameter(Mandatory = $false)] [ValidateSet("Present","Absent")] 
+                                        [String]   $Ensure = "Present",
+        [parameter(Mandatory = $false)] [String]   $Version = "1.0.0.0",
+        [parameter(Mandatory = $false)] [Boolean]  $Deployed = $true,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    $CurrentValues = Get-SPFarmSolutionInformation @PSBoundParameters
+    Write-Verbose -Message "Testing solution $Name"
+
+    $PSBoundParameters.Ensure = $Ensure
+
+    if ($WebApplications.Count -gt 0){
+        $valuesToCheck = @("Ensure", "Version", "Deployed", "WebApplications")
+    }else{
+        $valuesToCheck = @("Ensure", "Version", "Deployed")
+    }
+
+    return Test-xSharePointSpecificParameters -CurrentValues $CurrentValues -DesiredValues $PSBoundParameters -ValuesToCheck $valuesToCheck
+}
+
+function WaitFor-SolutionJob
+{
+    [CmdletBinding()]
+    param
+    (
+        [string]$SolutionName
+    )
+
+    start-sleep -s 5
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments @{ Name = $SolutionName } -ScriptBlock {
```

Comment:
Have you tested that this works correctly? You aren't passing $InstallAccount through to this function, so you might need to add that (I suspect it's running correctly with that being null and PowerShell 5 being present)